web/analysis.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+$dir = dirname(__FILE__);
+require_once "../conf/general";
+require_once "../commonlib/phplib/utility.php";
+
+$token = get_http_var('token');
+$file = basename(get_http_var('file'));
+
+function not_found() {
+  http_response_code(404);
+  include('404.html');
+  die();
+}
+
+if (!$token || !defined('OPTION_DATA_TOKEN') || !$file || $token != OPTION_DATA_TOKEN) {
+  not_found();
+}
+
+if (defined('OPTION_EXPORT_DATA_DIR')) {
```

Comment:
if not this, call not_found() first, or add this check to one above?

Reply:
yup. thanks.